packages/aws-cdk-lib/aws-cloudfront/lib/private/cache-behavior.ts

```diff
@@ -27,6 +27,10 @@ export class CacheBehavior {
   constructor(originId: string, private readonly props: CacheBehaviorProps) {
     this.originId = originId;
 
+    if (props.enableGrpc && props.allowedMethods !== AllowedMethods.ALLOW_ALL) {
+      throw new Error(`'allowedMethods' can only be AllowedMethods.ALLOW_ALL if you set 'enableGrpc' to true, got: ${props.allowedMethods?.methods}`);
+    }
```

Comment:
Otherwise CFn will fail:

```
Resource handler returned message: "Invalid request provided: AWS::CloudFront::Distribution: The parameter gRPC only supports the POST method. (Service: CloudFront, Status Code: 400, Request ID: ...
```